### PR TITLE
Rib 90 fix image upload

### PR DESCRIPTION
**Pull request by BenediktVitek**
- Pictures now loaded using google API. 
- Showing only last uploaded picture (on page view and on interest view)
- If no picture is connected to place, loading random image

### CODE REVIEW

**Review by dondowoldo — APPROVED** [review]

**Review by vdrfg — APPROVED** [review]

File: src/main/resources/templates/interest/page.html [hunk 1]
```diff
@@ -66,7 +66,12 @@ <h2>Places</h2>
                    th:href="@{/interests/{intId}/places/{placeId}(intId=${interest.id}, placeId=${place.id})}"
                 >
                     <div class="interest-place-img">
-                        <img src="https://picsum.photos/400/300" alt="placeholder"/>
+
+                        <img th:if="${place.place().imageNames.isEmpty()}" src="https://picsum.photos/400/300"
+                             alt="placeholder"/>
+                        <img th:unless="${place.place().imageNames.isEmpty()}" id="picture-container"
+                             th:attr="data-image-id=${place.place().imageNames.get(place.place().getImageNames().size() - 1)}"
+                             alt="placeholder">
```
Comment (marthubner) [inline src/main/resources/templates/interest/page.html:74]: Probably not a placeholder anymore

File: src/main/resources/templates/interest/page.html [hunk 2]
```diff
@@ -116,5 +121,7 @@ <h4 th:text="${place.address}"></h4>
         </section>
     </div>
 </div>
+<script th:inline="javascript" src="/scripts/imageFetcher.js"></script>
```
Comment (marthubner) [inline src/main/resources/templates/interest/page.html:124]: maybe th:inline="javascript" is redundant here?

File: src/main/resources/templates/place/page.html [hunk 3]
```diff
@@ -100,16 +99,20 @@ <h2>Rating criteria</h2>
 <script th:inline="javascript">
     /*<![CDATA[*/
 
+<script th:inline="javascript" src="/scripts/rating.js"></script>
```
Comment (marthubner) [inline src/main/resources/templates/place/page.html:102]: Broken whole <script> section after merge, should be probably:

```
<script th:replace="~{fragments :: placeMap-script}"></script>
<script th:inline="javascript">
    /*<![CDATA[*/

    let latitude = [[${place.latitude}]];
    let longitude = [[${place.longitude}]];
    let name = [[${place.name}]];
    let rating = [[${place.getAverageRating()}]];

    /*]]>*/
</script>
<script src="/scripts/placeMap.js"></script>
<script src="/scripts/imageFetcher.js"></script>
<script th:if="${usersRatings}" src="/scripts/rating.js"></script>
<script th:if="${usersRatings}" type="module" src="/scripts/imageUpload.js"></script>
```

File: src/main/resources/templates/interest/page.html [hunk 4]
```diff
@@ -66,7 +66,12 @@ <h2>Places</h2>
                    th:href="@{/interests/{intId}/places/{placeId}(intId=${interest.id}, placeId=${place.id})}"
                 >
                     <div class="interest-place-img">
-                        <img src="https://picsum.photos/400/300" alt="placeholder"/>
+
+                        <img th:if="${place.place().imageNames.isEmpty()}" src="https://picsum.photos/400/300"
```
Comment (marthubner) [inline src/main/resources/templates/interest/page.html:70]: Place.place().imageNames does not work anymore in latest develop branch after changes to PlaceInfoDTO.

List<String> imageNames needs to be added to PlaceInfoDTO for this to work.

**Review by marthubner — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/interest/page.html (hunk 2 above)
Comment (BenediktVitek) [reply src/main/resources/templates/interest/page.html]: fixed

File: src/main/resources/templates/place/page.html (hunk 3 above)
Comment (BenediktVitek) [reply src/main/resources/templates/place/page.html]: fixed

File: src/main/resources/templates/interest/page.html (hunk 4 above)
Comment (BenediktVitek) [reply src/main/resources/templates/interest/page.html]: Update PlaceInfoDTO and test using it

**Pull request closed without merging**